spec/salesforce_bulk/job_spec.rb

```diff
@@ -0,0 +1,247 @@
+  describe '#add_query' do
+    let (:records_or_query)  { "select Id, Company_ID__c, OwnerId, IsConverted from Lead where Company_ID__c != null limit 10" }
+    let (:xml_headers) { {"Content-Type" => "text/csv; charset=UTF-8"} }
+
+    it 'can adhere to the contract with Salesforce' do
+      instance = described_class.new(operation, sobject, records_or_query, external_field, connection)
+      described_class.class_variable_set(:@@job_id, job_id)
+
+      batch_id = "some-batch-id"
+
+      allow(connection).to receive(:post_xml).with(nil, "job/#{job_id}/batch/", records_or_query, xml_headers).and_return('<?xml version="1.0" encoding="UTF-8"?><batchInfo
+   xmlns="http://www.force.com/2009/06/asyncapi/dataload">
+ <id>' + batch_id + '</id>
+ <jobId>' + job_id + '</jobId>
+ <state>Queued</state>
+ <createdDate>2025-10-17T17:38:20.000Z</createdDate>
+ <systemModstamp>2025-10-17T17:38:20.000Z</systemModstamp>
+ <numberRecordsProcessed>0</numberRecordsProcessed>
+ <numberRecordsFailed>0</numberRecordsFailed>
+ <totalProcessingTime>0</totalProcessingTime>
+ <apiActiveProcessingTime>0</apiActiveProcessingTime>
+ <apexProcessingTime>0</apexProcessingTime>
+</batchInfo>')
+
+      instance = described_class.new(operation, sobject, records_or_query, external_field, connection)
+      expect(instance.add_query).to eq(batch_id)
+      expect(instance.instance_variable_get(:@@batch_id)).to eq(batch_id)
+    end
+  end
+
+  # https://developer.salesforce.com/docs/atlas.en-us.196.0.api_asynch.meta/api_asynch/asynch_api_batches_create.htm
+  describe '#add_batch' do
+    let (:operation) { "update" }
+    let (:records_or_query) { [
+      {"Company_ID__c": "C123456", "IsConverted": true},
+      {"Company_ID__c": "C234567", "IsConverted": true}
+    ] }
+    let (:xml_headers) { { "Content-Type" => "text/csv; charset=UTF-8" } }
+
+    it 'can adhere to the contract with Salesforce' do
+      instance = described_class.new(operation, sobject, records_or_query, external_field, connection)
+      described_class.class_variable_set(:@@job_id, job_id)
+      batch_id = "some-batch-id"
+
+      allow(connection).to receive(:post_xml).with(nil, "job/#{job_id}/batch/", "Company_ID__c,IsConverted\nC123456,true\nC234567,true\n", xml_headers).and_return('<?xml version="1.0" encoding="UTF-8"?>
+<batchInfo
+   xmlns="http://www.force.com/2009/06/asyncapi/dataload">
+ <id>' + batch_id + '</id>
+ <jobId>' + job_id + '</jobId>
+ <state>Queued</state>
+ <createdDate>2009-04-14T18:15:59.000Z</createdDate>
+ <systemModstamp>2009-04-14T18:15:59.000Z</systemModstamp>
+ <numberRecordsProcessed>0</numberRecordsProcessed>
+</batchInfo>')
+
+      instance = described_class.new(operation, sobject, records_or_query, external_field, connection)
+      expect(instance.add_batch).to eq(batch_id)
+      expect(instance.instance_variable_get(:@@batch_id)).to eq(batch_id)
```

Comment:
`@@batch_id` is a class variable
```suggestion
      expect(instance.class.class_variable_get(:@@batch_id)).to eq(batch_id)
```